File: 2864/MaximumOddBinaryNumber.cpp
/*
https://leetcode.com/problems/maximum-odd-binary-number/description/?envType=daily-question&envId=2024-03-01
*/

class Solution {
public:
    string maximumOddBinaryNumber(string s) {
        int countOnes=0, countZeros=0;
        for(char &ch: s){
            if(ch=='0') countZeros++;
            else countOnes++;
        }
        string res="";
        while(countOnes!=1){
            res+="1";
            countOnes--;
        }
        while(countZeros!=0){
            res+="0";
            countZeros--;
        }
        res+="1";
        return res;
    }
};

File: 2/2000s/28s/2864/MaximumOddBinaryNumber.java
/*
https://leetcode.com/problems/maximum-odd-binary-number/description/?envType=daily-question&envId=2024-03-01
*/

class Solution {
    public String maximumOddBinaryNumber(String s) {
        int countOnes=0, countZeros=0, sz=s.length();
        for(int i=0; i<sz; ++i){
            if(s.charAt(i)=='0') countZeros++;
            else countOnes++;
        }
        StringBuilder res=new StringBuilder();
        while(countOnes!=1){
            res.append("1");
            countOnes--;
        }
        while(countZeros!=0){
            res.append("0");
            countZeros--;
        }
        res.append("1");
        return String.valueOf(res);
    }
}